plugins/catalog-backend/src/ingestion/processors/microsoftGraph/read.ts

```diff
@@ -155,14 +158,17 @@ export async function readMicrosoftGraphGroups(
       kind: 'Group',
       metadata: {
         name: name,
-        description: group.displayName,
+        description: group.description ?? undefined,
```

Comment:
is this to also make null into undefined?

Reply:
Also checking - does this really validate? I thought that `k: undefined` is not supported, as opposed to the absence of `k`.

Reply:
The issue here is that the typings that are used here is using both `undefined` and `null`, however our entity definition only allows `undefined`. This validates correctly, the type is `description?: string | undefined`. I'm not a typescript expert, but I think `?` implies that `undefined` is allowed.

Reply:
Not quite, the question mark allows absence, the `| undefined` part allows the actual value to be `undefined` when the key is present - so they are mutually exclusive.

The actual `EntityMeta` definition in `catalog-model` says `description?: string;`, which would forbid undefined. So both here and in the profile you have to take care to not put undefined/null as values.

This is actually basically the entire reason that the ldap code has the awkward `mapStringAttr` etc stuff - to make it easier to just skip over missing values in a oneliner.

Reply:
I see what you mean, but neither eslint or vscode seem to behave that way. I have a look later and refactor it to avoid it 👍

Reply:
@freben Moved the code below and use a conditional assignment.